charts/backbeat/templates/consumer/deployment.yaml

```diff
@@ -1,26 +1,26 @@
 apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
-  name: {{ template "backbeat-consumer.fullname" . }}
+  name: {{ template "backbeat.fullname" . }}-consumer
   labels:
-    app: {{ template "backbeat-consumer.name" . }}
-    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    app: {{ template "backbeat.name" . }}
+    chart: {{ template "backbeat.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
-  replicas: {{ .Values.replicaCount }}
+  replicas: {{ .Values.consumer.replicaCount }}
   template:
     metadata:
       labels:
-        app: {{ template "backbeat-consumer.name" . }}
+        app: {{ template "backbeat.name" . }}
         release: {{ .Release.Name }}
     spec:
       containers:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           command: ["/usr/src/app/docker-entrypoint.sh"]
-          args: ["npm", "run", "queue_processor", "zenko"] 
+          args: ["npm", "run", "queue_processor", "zenko"]
```

Comment:
Is the expectation that we run multiple containers within the same chart (looking for direction) - one for `queue_populator` which produces entries into Kafka and `queue_processor` which consumes entries from Kafka and writes to MongoDB.

Reply:
Instead of having 2 charts (`backbeat-producer`, `backbeat-consumer`) each creating one `Deployment` which in turn runs one `Pod` with one container, there's now one chart (`backbeat`) which on install creates two `Deployment`s each running one `Pod` with one container.

Every `Deployment` is self-standing, e.g. changing the number of `consumer` replicas doesn't (and shouldn't) impact the number of `producer` replicas.

Rationale: there's no use in deploying `backbeat-consumer` without `backbeat-producer` and vice-versa, so combining them and deploying them as a single unit makes sense.

Keep in mind Charts have no direct relation to what *runs* in a cluster! E.g. the Prometheus chart will create a whole load of `Deployment`s, `DaemonSet`s and whatnot.

Reply:
There has been some discussions over the last two days (I am hoping that you were in the loop as well). We are working on building charts by use cases (lifecycle, replication, ingestion etc, backbeat api etc.). Keeping that in mind, I think this chart should start the API server which serves healthchecks and CRR metrics routes (for now, we will expand this provide more functionality).

Reply:
@rahulreddy I am, partially at least, but at the same time think this is not at all the right direction.

Nonetheless, it doesn't change the validity of this PR given state of things *right now*, so I'd like to move forward and merge.

Reply:
I agree, let's merge this.